src/lib/features/like-episode/model/like.ts

```diff
@@ -1,39 +1,57 @@
-import type { Episode } from '$lib/shared/api';
+import { supabaseClient, type Episode } from '$lib/shared/api';
 import { persistentWritable, localStorageAdapter } from 'svelte-persistent-writable';
+import { user } from '$lib/entities/user';
+
+user.subscribe(async ($user) => {
+  if (!$user) return;
+  const likes = await supabaseClient
+    .from('favourites')
+    .select('episode_id')
+    .order('created_at', { ascending: false });
+  const ids = likes.data?.map((e) => e.episode_id);
+  likesStore.set(new Set(ids ?? []));
+});
 
 /**
  * The store containing set of episodes liked by the current user.
  */
-export const likesStore = persistentWritable(new Map<Episode['id'], Episode>(), {
+export const likesStore = persistentWritable(new Set<number>(), {
   storage: localStorageAdapter('likes', {
-    serialize(likes: Map<Episode['id'], Episode>) {
-      const episodes: Episode[] = Array.from(likes.values());
-      return JSON.stringify(episodes);
+    serialize(likes: Set<number>) {
+      return JSON.stringify([...likes]);
     },
     deserialize(serialized: string) {
-      const episodes: Episode[] = JSON.parse(serialized);
-      return new Map(episodes.map((episode) => [episode.id, episode]));
+      const ids: number[] = JSON.parse(serialized);
+      return new Set(ids);
     },
   }),
 });
 
-/**
- * Register that the user liked the given episode.
- * @param episode The id of the episode to be liked.
- */
-export function addLike(episode: Episode) {
-  likesStore.update((likes) => likes.set(episode.id, episode));
+async function addCloudLike(episodeId: number) {
+  try {
+    await supabaseClient.from('favourites').insert([
+      {
+        // eslint-disable-next-line camelcase
+        episode_id: episodeId,
+      },
+    ]);
+  } catch {
+    return {
+      status: 500,
+      error: 'Server error. Check internet connection and ensure you are signed in.',
+    };
+  }
 }
 
-/**
- * Remove the like for the given episode.
- * @param episode The id of the episode whose like is to be removed.
- */
-export function removeLike(episode: Episode) {
-  likesStore.update((likes) => {
-    likes.delete(episode.id);
-    return likes;
-  });
+async function deleteCloudLike(episodeId: number) {
+  try {
+    await supabaseClient.from('favourites').delete().eq('episode_id', episodeId);
+  } catch {
+    return {
+      status: 500,
+      error: 'Server error. Check internet connection and ensure you are signed in.',
```

Comment:
question: where is this handled/displayed?

Reply:
Added `alert(e.error)` calls.